notebooks/00-release-28.0.0/03-full-timeline-queries.ipynb

```diff
@@ -0,0 +1,584 @@
+    "def stop_streaming_job( target_table: str, reset_offsets: bool = False):\n",
+    "    print(f'Pause streaming ingestion: [{druid.rest.post(f\"/druid/indexer/v1/supervisor/{target_table}/suspend\",\"\", require_ok=False)}]')\n",
+    "    \n",
+    "\n",
+    "    tasks = druid.tasks.tasks(state='running', table=target_table)\n",
+    "    tasks_done = 0\n",
+    "    while tasks_done<len(tasks):\n",
+    "        tasks_done = 0\n",
+    "        clear_output( wait=True)\n",
+    "        print(f'Waiting for running tasks to publish their segments ...')\n",
+    "        for task in tasks:\n",
+    "            status = druid.tasks.task_status(task['id'])\n",
+    "            print(f\"Task [{task['id']}] Status:{status['status']['statusCode']} RunnerStatus:{status['status']['runnerStatusCode']}\")\n",
+    "            if (status['status']['statusCode']!='RUNNING'): \n",
+    "                tasks_done += 1 \n",
+    "        time.sleep(1)\n",
+    "            \n",
+    "    if reset_offsets:\n",
+    "        print(f'Reset offsets for re-runnability: [{druid.rest.post(f\"/druid/indexer/v1/supervisor/{target_table}/reset\",\"\", require_ok=False)}]')\n",
+    "    print(f'Terminate streaming ingestion: [{druid.rest.post(f\"/druid/indexer/v1/supervisor/{target_table}/terminate\",\"\", require_ok=False)}]')\n",
+    "\n",
+    "# Remove table data and metadata from Druid\n",
+    "def drop_table( target_table: str):\n",
+    "    # mark segments as unused \n",
+    "    druid.datasources.drop(target_table)\n",
+    "    # remove segment metadata and data for unused segments\n",
+    "    headers = {'Content-Type': 'application/json'}\n",
+    "    kill_task = {\n",
+    "      \"type\": \"kill\",\n",
+    "      \"dataSource\": target_table,\n",
+    "      \"interval\" : \"2000-09-12/2999-09-13\"\n",
+    "    }\n",
+    "    print(druid.rest.post(f\"/druid/indexer/v1/task\", json.dumps(kill_task),require_ok=False, headers=headers))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "472589e4-1026-4b3b-bb79-eedabb2b44c4",
+   "metadata": {},
+   "source": [
+    "## Generate history\n",
+    "Run the following cell to create and load 3 months of history up to midnight last night.\n",
+    "The data is ingested from this generated data file using SQL Based ingestion.\n",
+    "\n",
+    "When completed, you'll see a description of the final table.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f52a94fb-d2e4-403f-ab10-84d3af7bf2c8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# generate 90 days of click data\n",
+    "days_of_history = 90\n",
+    "\n",
+    "start_time = datetime.now()\n",
+    "start_time = start_time - timedelta(days=days_of_history)\n",
+    "start_date = start_time.strftime('%Y-%m-%dT%H:%M:%S.001')\n",
+    "print(f\"Starting to generate history at {start_date}.\")\n",
```

Comment:
Just came back to this again after finishing the review and your note on it taking a while to get to the API ... would it be possible to ingest an existing example dataset, rather than generating it, and to then just set up the datagen so that it creates a "live" version of it?  I could see that being helpful in a lot of situations going forward - ie, yeah you get the one out of the box, but the data generator gives us that in live mode.

Just thinking of options to keep up the momentum through the notebook :)

Reply:
the generation here is very fast, the comment about the delay is really about the time it takes the real-time ingestion to kick in... nothing we can do about that.. The datagen server already has that capability and I agree that in other scenarios we might want to do that, but I don't think it will help anything here.